Convert more net tests to ztest

punitvara opened the pull request.

punitvara committed:
tests: zoap: convert legacy test to ztest

This patch makes legacy test case to use ztest apis.

Signed-off-by: Punit Vara <punit.vara@intel.com>

punitvara committed:
tests: zoap: increase stack size

This patch avoids failure of test case after converted to ztest by
increasing stack size

Signed-off-by: Punit Vara <punit.vara@intel.com>

punitvara committed:
tests: trickle: convert legacy test to ztest

This patch makes legacy test to use ztest APIs.

Signed-off-by: Punit Vara <punit.vara@intel.com>

nashif commented:
"normal test" -> "legacy test"

lpereira commented on tests/net/lib/zoap/src/main.c:
```diff
@@ -64,61 +65,41 @@ static struct sockaddr_in6 dummy_addr = {
-		TC_PRINT("Could not get packet from pool\n");
-		goto done;
-	}
+	zassert_not_null(pkt, "Could not get packet from pool");
```
Messages in assertions should be written as if they were true: `zassert_not_null(pkt, "Get packet from pool")`. If `pkt` is `NULL`, ztest will then tell you that `the "get packet from pool" assertion failed`, which makes more sense. Otherwise, it's a double negative.

punitvara replied:
Done. Thanks for the review

punitvara committed:
tests: context: Make legacy test to use ztest API

Use ztest APIs in test case to make output unified with other test
cases.

Signed-off-by: Punit Vara <punit.vara@intel.com>

punitvara committed:
tests: context: increase stack size

Increase stack size to avoid failure in qemu

Signed-off-by: Punit Vara <punit.vara@intel.com>

punitvara committed:
tests: route: Convert legacy test to use ztest framework

This pacth makes use of ztest apis to make output unified with other
test cases.

Signed-off-by: Punit Vara <punit.vara@intel.com>

punitvara committed:
tests: rpl: convert legacy test in ztest

This patch makes use of ztest apis to make output unified with
other test cases

Signed-off-by: Punit Vara <punit.vara@intel.com>

punitvara committed:
tests: tcp: convert legacy test to use ztest

This patch removes coding style issues and makes use of
ztest apis in legacy test case.

Signed-off-by: Punit Vara <punit.vara@intel.com>

nashif commented on tests/net/trickle/src/main.c:
```diff
@@ -79,111 +79,74 @@ static void cb_2(struct net_trickle *trickle, bool do_suppress,
-		TC_ERROR("Trickle 1 not running\n");
-		return false;
+		zassert_true(0, "Trickle 1 not running");
 	}
```
you should get rid of the if statement and do the checking in the zassert, same below

jhedberg commented on tests/net/context/src/main.c:
```diff
@@ -66,87 +67,60 @@ static struct k_sem wait_data;
 	if (net_context_is_used(context)) {
-		TC_ERROR("Context put check test failed.\n");
-		return false;
+		zassert_true(0, "Context put check test failed");
```
Wouldn't it make more sense to do zassert_false(net_context_is_used(context));

nashif replied:
agree

punitvara committed:
tests: fragment: convert legacy test to ztest

test makes use of ztest apis and coding style issue are
cleaned up.

Signed-off-by: Punit Vara <punit.vara@intel.com>

punitvara committed:
tests: 6lo: convert legacy test to ztest with cleanup

Test uses ztest and coding cleaup errors along with
warnings are removed through this patch

Signed-off-by: Punit Vara <punit.vara@intel.com>

nashif commented on tests/net/context/src/main.c:
```diff
@@ -786,58 +659,48 @@ static bool net_ctx_sendto_v4_wrong_src(void)
 	if (!k_sem_take(&wait_data, WAIT_TIME)) {
-		TC_ERROR("Semaphore triggered but should not\n");
-		return false;
+		zassert_true(0, "Semaphore triggered but should not");
```
can you check for the return directly in the assert?

punitvara replied:
Done

nashif commented on tests/net/context/src/main.c:
```diff
@@ -786,58 +659,48 @@ static bool net_ctx_sendto_v4_wrong_src(void)
-		TC_ERROR("Data received but should not have, "
-			 "IPv4 recv test failed\n");
-		return false;
+		zassert_true(0, "Data received but should not have, "
```
ditto

nashif commented on tests/net/context/src/main.c:
```diff
@@ -786,58 +659,48 @@ static bool net_ctx_sendto_v4_wrong_src(void)
-		TC_ERROR("No data received on time 2nd time, "
-			 "IPv6 recv test failed\n");
-		return false;
+		zassert_true(0, "No data received on time 2nd time, "
```
ditto

nashif commented on tests/net/context/src/main.c:
```diff
@@ -786,58 +659,48 @@ static bool net_ctx_sendto_v4_wrong_src(void)
-		TC_ERROR("No data received on time 2nd time, "
-			 "IPv4 recv test failed\n");
-		return false;
+		zassert_true(0, "No data received on time 2nd time, "
```
ditto

nashif commented on tests/net/context/src/main.c:
```diff
@@ -851,58 +714,48 @@ static void recv_cb_another(struct net_context *context,
-		TC_ERROR("Context recv reconfig IPv6 UDP test failed (%d)\n",
-			 ret);
-		return false;
+		zassert_true(0, "Context recv reconfig IPv6 UDP test failed");
```
ditto

nashif commented on tests/net/context/src/main.c:
```diff
@@ -851,58 +714,48 @@ static void recv_cb_another(struct net_context *context,
-		TC_ERROR("No data received on time, "
-			 "IPv6 recv reconfig test failed\n");
-		return false;
+		zassert_true(0, "No data received on time, "
```
ditto, and looks like many occurrences below

jukkar commented on tests/net/6lo/src/main.c:
```diff
@@ -165,8 +166,8 @@ static struct net_icmpv6_nd_opt_6co ctx2 = {
 static const char user_data[] =
-		"0123456789012345678901234567890123456789"
-		"0123456789012345678901234567890123456789"
+		"0123456789012345678901234567890123456789 "
```
Why are you adding space character to the data?

jukkar commented on tests/net/6lo/src/main.c:
```diff
@@ -352,8 +353,10 @@ static struct net_pkt *create_pkt(struct net_6lo_data *data)
 
-	/* length is not set in net_6lo_data data pointer, calculate and set
-	 * in ipv6, udp and in data pointer too (it's required in comparison) */
+/*
```
Indentation is wrong here, the original was correct

jukkar commented on tests/net/ieee802154/fragment/src/main.c:
```diff
@@ -101,34 +102,34 @@
-		"0123456789012345678901234567890123456789"
-		"0123456789012345678901234567890123456789"
-		"0123456789012345678901234567890123456789"
+		"0123456789012345678901234567890123456789 "
```
What is the reason for the extra space?

jukkar commented on tests/net/ieee802154/fragment/src/main.c:
```diff
@@ -244,8 +245,10 @@ static struct net_pkt *create_pkt(struct net_fragment_data *data)
 	len = NET_UDPH_LEN + remaining;
-	/* length is not set in net_fragment_data data pointer, calculate and set
-	 * in ipv6, udp and in data pointer too (it's required in comparison) */
+/*
```
Original indentation is correct

punitvara replied:
Threre was error from checkpatch thats why I changed this. But I will make something similar now

jukkar commented on tests/net/ieee802154/fragment/src/main.c:
```diff
@@ -296,6 +300,7 @@ static struct net_fragment_data test_data_1 = {
 	.iphc = true
 };
 
+
```
Extra empty line

jukkar commented on tests/net/tcp/src/main.c:
```diff
@@ -494,9 +494,9 @@ static bool send_ipv4_tcp_msg(struct net_if *iface,
-			DBG("Timeout, packet not received\n");
-			return false;
+
+		DBG("Timeout, packet not received\n");
```
Wrong indentation here

jukkar commented on tests/net/tcp/src/main.c:
```diff
@@ -545,9 +545,9 @@ static bool send_ipv6_tcp_long_msg(struct net_if *iface,
-			DBG("Timeout, packet not received\n");
-			return false;
+
+		DBG("Timeout, packet not received\n");
```
Indentation again

jukkar commented on tests/net/tcp/src/main.c:
```diff
@@ -666,84 +659,99 @@ static bool test_register(void)
 	})
 
 #define REGISTER_FAIL(raddr, laddr, rport, lport)			\
+	do {								\
```
Please fix indentation as you added do-while block, same comment to other macro changes below

jukkar commented on tests/net/tcp/src/main.c:
```diff
@@ -938,32 +943,28 @@ static bool test_create_v6_reset_packet(void)
 	if (!v6_check_port_and_address("TCP reset", pkt,
 				       &peer_v6_inaddr, PEER_TCP_PORT)) {
-		return false;
+	zassert_true(0, NULL);
```
Indentation is wrong

jukkar commented on tests/net/tcp/src/main.c:
```diff
@@ -1012,32 +1009,28 @@ static bool test_create_v6_syn_packet(void)
 	if (!v6_check_port_and_address("TCP syn", pkt,
 				       &peer_v6_inaddr, PEER_TCP_PORT)) {
-		return false;
+	zassert_true(0, NULL);
```
ditto

jukkar commented on tests/net/context/src/main.c:
```diff
@@ -66,87 +67,59 @@ static struct k_sem wait_data;
-		return false;
-	}
+	zassert_equal(ret, -EAFNOSUPPORT,
+			"Invalid family test failed");
```
Wrong indentation here and in many lines below, please fix them all

jukkar commented on tests/net/ieee802154/fragment/src/main.c:
```diff
@@ -415,29 +420,23 @@ static struct net_fragment_data test_data_8 = {
-		goto end;
-	}
+	zassert_true(net_6lo_compress(pkt, data->iphc,
+			      ieee802154_fragment),
```
Fix indentation here

jukkar commented on tests/net/lib/zoap/src/main.c:
```diff
@@ -186,67 +150,45 @@ static int test_build_simple_pdu(void)
-		goto done;
-	}
+	zassert_equal(frag->len, sizeof(result_pdu),
+			"Different size from the reference packet");
```
Indentation wrong here

jukkar commented on tests/net/lib/zoap/src/main.c:
```diff
@@ -255,99 +197,62 @@ static int test_build_no_size_for_options(void)
-		goto done;
-	}
+	zassert_equal(ver, 1,
+			"Invalid version for parsed packet");
```
Indentation wrong here, there seems to many lines below with same issue

jukkar reviewed (changes requested):
Lot of indentation issues and I probably missed many in the review. Please go through all the changes and make sure you have proper indentation.

punitvara commented:
Merge conflicts were taking time to solve. I have created new PRs for each test case. This PR will be closed once those PR get merged
